src/js/components/Courses/Courses.js

```diff
@@ -28,17 +31,34 @@ class Courses extends PureComponent {
       .catch(() => this.setState({ courses: [] }));
   }
 
+  handleClickMore = () => {
+    this.setState(prevState => ({ page: prevState.page + 1 }), () => {
+      getMoreData(this.state.page, limit)
```

Comment:
you can move getMoreData with all set states to function and use it on click and on did mount to avoid code duplication

Reply:
I think it's impossible because in componentDidMount and in handleClickMore there are different set states..

Reply:
r u sure?)

      .then(() => this.setState({ isFetching: false })); in both places, 
this.setState(prevState => ({ courses: [...prevState.courses, ...courses] })); and .then(courses => this.setState({ courses })) are the same because we still have an empty array in initial state

they r different in loadedData sesState and on click you do smth in catch, but if your function will return Promise it'll possible to continue the chain where you need